### PR TITLE
Bugfix/15508-stock-tools-measure-x-annotation

### LINKED ISSUE #15508: Annotation with measure type "x" does not work as expected
Hi there!
When I'll try to implement the annotation measure tools I'd realize that when I don't want to use the range selector config, the measure tool for "x" doesn't show the labels at all:
image

But if I put the rangeSelector enabled the behavior works as expected
image

#### Live demo with steps to reproduce
https://jsfiddle.net/fpmu4n2o/83/

#### Affected browser(s)
Google Chrome

am I not implementing it correctly?
Thanks in advance and good works anyway!

### PR DESCRIPTION
**Pull request by pawellysy**
Fixed #15508, the measure annotation's label wasn't correctly rendered inside the annotation.

### CODE REVIEW

File: samples/unit-tests/annotations/annotations-measure/demo.js [hunk 1]
```diff
@@ -29,37 +29,53 @@ QUnit.test('#13664 - annotation measure on yAxis', function (assert) {
...
+        }]
     });
 
     assert.ok(
         chart.annotations[0].shapesGroup.getBBox().y === chart.yAxis[1].top,
         'Annotation measure should be visible on vary yaxis (#13664).'
     );
+
+    assert.close(
+        chart.annotations[0].shapesGroup.getBBox().y,
```
Comment (karolkolodziej) [inline samples/unit-tests/annotations/annotations-measure/demo.js:46]: I would assign this to a variable in order to reduce the number of times when this method is fired.

File: ts/Extensions/Annotations/Types/Measure.ts [hunk 2]
```diff
@@ -19,10 +19,15 @@ import U from '../../../Core/Utilities.js';
 const {
     extend,
     isNumber,
-    merge
+    merge,
+    pick
 } = U;
 
 /* eslint-disable no-invalid-this, valid-jsdoc */
+interface ExtendedPositionObject extends PositionObject{
```
Comment (karolkolodziej) [inline ts/Extensions/Annotations/Types/Measure.ts:27]: @bre1470 could you please confirm if that is the optimal way of extending that interface?

File: ts/Extensions/Annotations/Types/Measure.ts [hunk 3]
```diff
@@ -498,22 +503,20 @@ class Measure extends Annotation {
...
+                point: function (target: any): ExtendedPositionObject {
+                    const annotation: Measure = target.annotation;
+                    const options = target.options;
 
                     return {
-                        x: (inverted ? top : 10) +
-                            xAxis.toPixels(annotation.xAxisMin, !inverted),
-                        y: (inverted ? -left + 10 : top) +
-                            yAxis.toPixels(annotation.yAxisMin)
+                        x: annotation.xAxisMin,
```
Comment (karolkolodziej) [inline ts/Extensions/Annotations/Types/Measure.ts:516]: IMHO the design without that extra 10px padding looks cleaner. 
If everyone agrees with that, please accept and add some comment on the visual difference.

File: ts/Extensions/Annotations/Annotations.ts [hunk 4]
```diff
@@ -165,6 +165,8 @@ declare global {
             point?: (string|MockPointOptions);
             itemType?: string;
             vertical?: VerticalAlignValue;
+            xAxis?: number;
```
Comment (karolkolodziej) [inline ts/Extensions/Annotations/Annotations.ts:168]: A few lines above- do we need this `extends` with an almost identical name?
When I merge that, everything works as expected, but maybe there is another reason for that?
``` TS
 interface AnnotationsLabelsOptions extends AnnotationsLabelOptions
```

**Review by karolkolodziej — CHANGES_REQUESTED** [review]

File: ts/Extensions/Annotations/Types/Measure.ts [hunk 5]
```diff
@@ -498,22 +503,20 @@ class Measure extends Annotation {
...
-                point: function (target: any): PositionObject {
-                    const annotation: Measure = target.annotation,
-                        chart = annotation.chart,
-                        inverted = chart.inverted,
-                        xAxis = chart.xAxis[typeOptions.xAxis],
-                        yAxis = chart.yAxis[typeOptions.yAxis],
-                        top = chart.plotTop,
-                        left = chart.plotLeft;
+                xAxis: 0,
+                yAxis: 0,
```
Comment (raf18seb) [inline ts/Extensions/Annotations/Types/Measure.ts:510]: Could you double-check if it's working on multiple panes?

File: ts/Extensions/Annotations/Types/Measure.ts [hunk 6]
```diff
@@ -498,22 +503,20 @@ class Measure extends Annotation {
...
+                point: function (target: any): ExtendedPositionObject {
+                    const annotation: Measure = target.annotation;
+                    const options = target.options;
```
Comment (raf18seb) [inline ts/Extensions/Annotations/Types/Measure.ts:513]: Would be nice to stick with our pattern of declaring multiple variables with only one `const` key word:
```js
const annotation: Measure = target.annotation,
    options = target.options;
```

**Review by raf18seb — CHANGES_REQUESTED** [review]
The label is correct when you create annotation by dragging the mouse from left to right, but it's not working when dragging from right to left. Could you check it?

File: ts/Extensions/Annotations/Types/Measure.ts (hunk 5 above)
Comment (pawellysy) [reply ts/Extensions/Annotations/Types/Measure.ts:510]: It does work in multiple panes 👍

File: ts/Extensions/Annotations/Annotations.ts (hunk 4 above)
Comment (pawellysy) [reply ts/Extensions/Annotations/Annotations.ts:168]: @bre1470 could you take a look?

File: samples/unit-tests/annotations/annotations-measure/demo.js [hunk 7]
```diff
@@ -29,37 +29,53 @@ QUnit.test('#13664 - annotation measure on yAxis', function (assert) {
...
+        chart.annotations[0].labels[0].graphic.anchorY,
+        0.5,
+        "Annotation's label's Y position should be close to the Y position of the annotation."
```
[the remark this answers is not in the document]
Comment (pawellysy) [reply samples/unit-tests/annotations/annotations-measure/demo.js:50]: I used double quotes, to use single quote as a punctuation mark in the name of the test. Isn't it desired solution?
@pawelfus

File: ts/Extensions/Annotations/Annotations.ts (hunk 4 above)
Comment (bre1470) [reply ts/Extensions/Annotations/Annotations.ts:168]: https://api.highcharts.com/highcharts/annotations.labelOptions is not identical to https://api.highcharts.com/highcharts/annotations.labels options.

File: ts/Extensions/Annotations/Types/Measure.ts (hunk 2 above)
Comment (pawellysy) [reply ts/Extensions/Annotations/Types/Measure.ts:27]: isn't it exactly this?
```js
{
x: number;
y: number;
xAxis?:number;
yAxis?:number;
}
```

**Comment by TorsteinHonsi** [comment]
@pawellysy , regarding the PR description, see [writing content for the changelog](https://github.com/highcharts/highcharts/blob/master/changelog/readme.md#writing-content-for-the-changelog), points 5 and 6 .

File: ts/Extensions/Annotations/Types/Measure.ts (hunk 2 above)
Comment (pawelfus) [reply ts/Extensions/Annotations/Types/Measure.ts:27]: `x/yAxis?: number|string;` should be correct too. In theory `xAxis: chart.xAxis[index]` is also supported, but I have never tested it

I'm not sure if `ExtendedPositionObject` name is correct here, sorry for the confusion. Just checked, we already have `MockPointOptions` interface defined, let's reuse it

**Review by pawelfus — CHANGES_REQUESTED** [review]

**Review by karolkolodziej — APPROVED** [review]
Please fix the TS and accept visual differences.

**Review by raf18seb — APPROVED** [review]
Thanks!

File: ts/Extensions/Annotations/Annotations.ts [hunk 8]
```diff
@@ -165,6 +165,8 @@ declare global {
...
+            yAxis?: number;
```
Comment (pawelfus) [inline ts/Extensions/Annotations/Annotations.ts:169]: Could you change it to: `yAxis?: number|string`? The same for `xAxis`

**Review by pawelfus — CHANGES_REQUESTED** [review]
Sorry, I missed this in the previous review, could you fix this minor issue? See below
Thanks!

**Review by pawelfus — APPROVED** [review]
Thanks! :)

**Review by TorsteinHonsi — APPROVED** [review]

**Pull request merged**